packages/wallet/src/hasher.ts

```diff
@@ -11,6 +16,51 @@ export function hashMessage(msg: string) {
   return sha256(Buffer.from(msg));
 }
 
+/**
+ * Hash transaction request with sha256. [Read more](https://github.com/FuelLabs/fuel-specs/blob/master/specs/protocol/identifiers.md#transaction-id)
+ *
+ * @param transactionRequest - Transaction request to be hashed
+ * @returns sha256 hash of the transaction
+ */
+export function hashTransaction(transactionRequest: TransactionRequest) {
+  // Return a new transaction object without references to the original transaction request
+  const transaction = transactionFromRequest(transactionRequest);
+
+  if (transaction.type === TransactionType.Script) {
+    transaction.receiptsRoot = '0x0000000000000000000000000000000000000000000000000000000000000000';
+  }
+
+  // Zero out input fields
+  transaction.inputs = transaction.inputs.map((input) => {
+    const inputClone = { ...input };
+
+    switch (inputClone.type) {
+      // Zero out on signing: txoPointer
+      case InputType.Coin: {
+        // inputClone.txoPointer = 0;
```

Comment:
If I understand correctly if the transaction's input is a coin, shouldn't we set the `txoPointer` to `0` when signing? Why is this commented out?

Reply:
We didn't have the field `txoPointer` yet on the transaction.

Reply:
Oh I see, makes sense then.